tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,249 @@
+                   'action_flag': ADDITION,
+                   'object_repr': "Clean and Re-parse"
+                   }
+        return context
+
+    def __backup(self, backup_file_name, log_context):
+        """Execute Postgres DB backup."""
+        try:
+            logger.info("Beginning re-parse DB Backup.")
+            call_command('backup_db', '-b', '-f', f'{backup_file_name}')
+            logger.info("Backup complete! Commencing clean and re-parse.")
+
+            log("Database backup complete.",
+                logger_context=log_context,
+                level='info')
+        except Exception as e:
+            log("Database backup FAILED. Clean and re-parse NOT executed. Database and Elastic are CONSISTENT!",
+                logger_context=log_context,
+                level='error')
+            raise e
+
+    def __handle_elastic(self, new_indices, delete_indices, log_context):
+        """Create new Elastic indices and delete old ones."""
+        if new_indices:
+            try:
+                if not delete_indices:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias', '--use-alias-keep-index')
+                else:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias')
+                log("Index creation complete.",
+                    logger_context=log_context,
+                    level='info')
+            except Exception as e:
+                log("Elastic index creation FAILED. Clean and re-parse NOT executed. "
+                    "Database is CONSISTENT, Elastic is INCONSISTENT!",
+                    logger_context=log_context,
+                    level='error')
+                raise e
+            # TODO: Need to ask Alex if we can run some queries in prod to deduce the average number of records per DF.
+
+    def __delete_records(self, docs, file_ids, new_indices, log_context):
+        """Delete records, errors, and documents from Postgres and Elastic."""
+        total_deleted = 0
+        self.__delete_errors(file_ids, log_context)
+        for doc in docs:
+            # atomic delete?
+            try:
+                model = doc.Django.model
+                qset = model.objects.filter(datafile_id__in=file_ids)
+                total_deleted += qset.count()
+                if not new_indices:
+                    # If we aren't creating new indices, then we don't want duplicate data in the existing indices.
+                    doc().update(qset, refresh=True, action='delete')
+                qset._raw_delete(qset.db)
+            except ElasticsearchException as e:
+                log(f'Elastic document delete failed for model {model}. Database and Elastic are INCONSISTENT! Restore '
+                    'the DB from the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+            except Exception as e:
```

Comment:
For readability, my suggestion is to write an exception for raw query instead of using generic Exception. Using Exception should be the last resort

Reply:
I do disagree with this in this case. I think we need to catch the lowest possible exception case to ensure we notify the user that a failure occurred and the DB could be in an inconsistent state. Im open to adding more exceptions to the `except` block, but I think every single `except` block should be catching `Exception` as a safety net.

Reply:
I added some slightly more specific exception handling. Let me know what you think!